### DISCUSSION TITLE
Replacing grammars where the language has spaces in its name

### DISCUSSION THREAD

**Discussion by TheWitheredStriker** [github-linguist/linguist#6019] [Q&A] [answered]
I tried replacing the grammar of GML (the scripting language of GameMaker) as it currently uses a C grammar rather than its own. However, GML is listed in `languages.yml` as `Game Maker Language`, with whitespace. Because of this I presume the command to replace the grammar isn't working:

![image](https://user-images.githubusercontent.com/38614982/184613305-e6f93745-5ef6-44eb-bd98-c42a0a5db154.png)

How would I go about this? Am I using the command correctly? The command says `script/add-grammar --replace MyGrammar https://github.com/PeterPan/MyGrammar`, which made me think that maybe I need to specifiy a grammar instead of a language, but wouldn't specifiying `c.tmbundle` replace it everywhere, including in the actual C language? Sorry for the dumb questions, this is my first time contributing to Linguist.

**Comment by lildude** [comment, marked as answer]
Normally, you'd use the name as it appears in the directory structure under `vendor/grammars` however in this case as it's sharing the C grammar, you need to add the grammar as a new grammar rather than replace it.

**Reply by TheWitheredStriker** [reply]
Having some issues with my Ubuntu VM at the moment; I'll try this as soon as I get it working again and will then reply here if it worked or not.

**Reply by TheWitheredStriker** [reply]
Due to the ongoing problems with my Ubuntu, a friend of mine is helping me out with this using his Linux. [My grammar file's repository](https://github.com/TheWitheredStriker/gml-support) was successfully added to our fork of Linguist, but [Linguist still warns that it hasn't found any grammar files](https://github.com/TheWitheredStriker/linguist/pull/1/commits/28b93601e505107b31d5ae38ad073efca465c9a5). What do we do now?

**Reply by lildude** [reply]
Your grammar files need to be in specific directories depending on the extension in order for the compiler to find them:

https://github.com/github/linguist/blob/e78ef71af3600f96b9a40a06511ebbe3797e7401/tools/grammars/compiler/loader.go#L119-L128

**Reply by TheWitheredStriker** [reply]
Ah, I see. So `syntaxes/main/...` doesn't cut it? I forked the repository, which is why that structure was already there, but I'm fine with changing it.

Guessing the `gml` and `gmx` folders also need to go?
